### DISCUSSION TITLE
Problem with ripping TV episodes

### DISCUSSION THREAD

**Discussion by HaroldRobi** [automatic-ripping-machine/automatic-ripping-machine#594] [General]
If I have another open discussion, please feel free to delete it.  I can not see it in the list of discussions, and I wanted to provide more information.

Good Morning.

I am having a problem ripping Tv shows.  It seems that Automatic ripper is not ripping all the episodes.  Its only ripping one.  For example CSI: Crime Scene Investigatoin, the complete 3rd season on DVD.  If I have ARM rip Season 3 Disk 6.  It will only rip 1 episode.  It will create a folder called "DVD Video"  in thiat folder it will create a file called "DVD_Video.mkv" and then stop after the first track it rips.

Yet, If I open this DVD in MakeMKV in Windows, you will plainly see there are 3 episdes on the disk, and has way more information than "DVD_Video.mkv"

For example.  The first episode on this disk is called CSI: Season 3, Disk-6 - 6 chapters, 1.7GB (A1)
The second episode is called The first episode on this disk is called CSI: Season 3, Disk-6 - 6 chapters, 1.7GB (A2)
The third episdoe is called: The first episode on this disk is called CSI: Season 3, Disk-6 - 6 chapters, 1.7GB (A3)

I have the entire CSI Crime Scene Investigaion series, and was blowing through them.  I thought "wow ARM is fun"  Now I have a bunch of garbage I need to delete and re-rip because its useless.
When I rip the DVD in windows, MakeMKV makes a folder called "CSI-Season3" the name of the file is "Disc 6-A1_T00.mkv

It appears that you might be missing some information when creating these files.

Also as a second question,  Can you make it easer to update the MakeMKV license?  Pleace this option in ArmOptions menu?

Regards,

Harold Robinson

**Comment by HaroldRobi** [comment]
After looking into this issue further, I found an "extras" folder where the rest of the episodes reside.  Inside the extras folder those files are named correctly  Its just the folder and the first episode that was ripped is named DVD_Video.

**Reply by microtechno9000** [reply]
ARM at the moment does have some issues when ripping TV Series. I have also found it rips as if it was a movie creating a single feature with all episodes in one, and sometimes putting the single episodes into the extra's folder.

Once I get some spare time end of November, I am keen to start working on this problem and will hopefully have a fix for some of the TV, Movie and other media ripping issues.

**Reply by HaroldRobi** [reply]
count me in.  be happy to help

**Comment by gibby82** [comment]
Apologies to revive an old thread (point me to a more recent one if it exists). I'm seeing the exact same behavior (The Expanse) regarding TV disc ripping. Was there any progress made? 

So far I have observed all of the items Harold mentioned - the extras folder, the 'long' episode, and also I have noticed missing episodes. 

In any case, across two shows I've definitely noticed it is still an issue. Let me know if more info is needed or if any testing is required. I've got 2-3 box sets to get through.

**Comment by nickredible** [comment]
I've observed the same issue, including the missing episodes, whilst ripping a box set of Only Fools and Horses. First two seasons were ripped with the correct number of episodes, but later seasons are missing the last 2-3 episodes. If there's anything I can help with, I'd be happy to do so.

**Comment by TheSniperM0nkey** [comment]
I have the same issue with ripping episodes. I don’t mind them going into the extras folder, but out of three new release discs two ripped all episodes and one had one missing. I previously assumed my older dvd series were just too old which was the cause of the issue but seeing this it is probably the same issue seen here as they play without issue.

**Comment by ember1205** [comment]
Have you tried turning the logging level up and reading through the details? I know that with movies, after it rips all of the various files, it checks to see which is the "largest", uses that one as the Main Feature, and everything else gets dumped in the Extras folder. I'm wondering if what is happening here with TV Episodes is effectively the same thing.

As far as some episodes still actually being missing, have you checked the sizes of the files and actually looked at the content? Sometimes, two episodes are actually contained in a single file. Check file sizes and the run-time length of the episodes to see if what you're "missing" is actually part of another episode file.

**Comment by TheSniperM0nkey** [comment]
Yeah that’s exactly what happens.

In my experience the missing episodes aren’t combined or anything like that, they are all single episode files it’s just missing some.

**Comment by ember1205** [comment]
I would be looking at the logs... 

Friends (complete series, BluRay) has two episodes in Season 9 (Episodes 23 and 24) combined together. As a result, the file gets named for only one of the episodes, but the file size is roughly double that of all of the one-episode-per-file episodes.

A lot of TV shows, especially older ones, are "fakes" and not actually correctly mastered in the first place. While they will typically play fine in a DVD player, all it takes is one "bad bit" in the data to cause it to not rip. I ran into a TON of trouble with Everybody Loves Raymond - I had originally bought the entire series in a box set and had issues with about 25-30 episodes not ripping because of disc errors (it was a counterfeit set of discs).

You would definitely want to be looking at the rip logs to see what the system believes is on the disc and what it's actually able to rip and go from there.

**Comment by gibby82** [comment]
I will attempt to dig through logs when I get time. However, I have noticed that it would seem that ARM, while technically capable of ripping TV media, isn't setup for success. 

1. The lookup DBs seem to be movie oriented only. This results in poor matches for TV a majority of the time. 
2. The TV media automatic ripping behavior is all over the place, likely due to the variance of TV track layouts, but likely also I suspect due to it looking for movie track layouts.

In a few cases I've simply given up on ripping TV with ARM _or_ manually because it is quite a tedious endeavor. The Expanse for example - episodes would not rip in a consistent order, so I had to spend a ton of time trying to manually identify which episode was which (and failing at it). 

So it might just be that ARM isn't the right tool for TV, or it needs a lot of attention to become the right tool. Not sure which.

**Comment by HaroldRobi** [comment]
I have personally ripped many TV shows using ARM. This is not an automatic
process. I just rip the disks select the TV show it's ripping (which gets
the right folder and I start ripping. I then have to go back through every
file to see what episode it is, name it correctly.  Sometimes the DVDs also
have a watch all video file which will be larger than the standard video
file which gets deleted. I rinse and repeat. I did Bewitched in 3 days.  I
used TheTvDB to name the episodes correctly

So yes it's possible but a lot of manual intervention is required.

Regards

-------------
"We've all seen that bumper sticker:    If God rode a motorcycle, it would
be a Harley.  I thought to myself...... now I know why God
hasn't made His second coming.....  He's still stuck in a cornfield out in
Nebraska or somewhere waiting for parts!"

**Reply by ember1205** [reply]
> I have personally ripped many TV shows using ARM. This is not an automatic process. I just rip the disks select the TV show it's ripping (which gets the right folder and I start ripping. I then have to go back through every file to see what episode it is, name it correctly. Sometimes the DVDs also have a watch all video file which will be larger than the standard video file which gets deleted. I rinse and repeat. I did Bewitched in 3 days. I used TheTvDB to name the episodes correctly So yes it's possible but a lot of manual intervention is required. Regards
> 

Agree with this sentiment... The actual ripping of content is generally ok so long as ARM is configured to rip the full set if tracks. What ARM actually does with those files is another matter. It may do it's "this is the largest file, so it's the main title and everything else goes in Extras" routine like it does with movies, so just be aware of that.

I built a brand new system to build this new ARM on top of, and it worked great for about a dozen movies. Since then, it has puked all over itself at every turn, frustrating me to no end.  Combine that with how many things don't actually work (you can't actually designate a drive for manual use - it's ALWAYS automatic - is just one example) and I've taken to just executing the makemkvcon command myself and I get very reliable results that way.

**Reply by gibby82** [reply]
@HaroldRobi What I've found is that what you described is a best case scenario. I've done this with Gilmore Girls for example, works just fine. In this case the files are at least labeled with a discernible numerical order. In the case of the Expanse, the labels were flat out wrong, which made episode renaming quite difficult. I had to compare episode summaries to what I was watching to match an episode. 

I am also using Filebot for some of the renaming process which does have TVDB as a lookup. However ARM doesn't have this option that I'm aware off, which is unfortunate. 

So yes, it isn't automatic, but that is at odds with the intended use case. :)

**Reply by ember1205** [reply]
> @HaroldRobi What I've found is that what you described is a best case scenario. I've done this with Gilmore Girls for example, works just fine. In this case the files are at least labeled with a discernible numerical order. In the case of the Expanse, the labels were flat out wrong, which made episode renaming quite difficult. I had to compare episode summaries to what I was watching to match an episode.
> 
> I am also using Filebot for some of the renaming process which does have TVDB as a lookup. However ARM doesn't have this option that I'm aware off, which is unfortunate.
> 
> So yes, it isn't automatic, but that is at odds with the intended use case. :)

ARM can use OMDB (movies and TV shows) or TMDB (focus on movies only). Whichever one you opt for, you need to create an API key for your use and add it to the configuration. 

For me, the broader issue is that ARM does a poor job of naming in the first place. Even though the name coded directly onto the disc was fairly close to begin with, it still completely whiffed at the Ocean's trilogy, half of the Fast and Furious franchise, and about half of the Jurassic Park franchise. And even when it got the right name, the automatic naming insists on putting dashes in between the words in the title but forcing a title search manually afterward using the guessed info returns a name WITHOUT the dashes... This makes absolutely ZERO sense to me as it seems that it's using two completely different routines to do searches and return values.

TV Shows are just generally harder because a single disc typically covers either an entire season or worse... only PART of a season. And file names are numbered starting at zero which doesn't when you're trying to rip Season 06 Episodes 07 through 12...

Filebot is a decent tool, but I have found it works best when you're at least starting with SOME level of accuracy to the files already. I know it always did a good job for me for movies, but I don't really remember using it for TV Shows so I can't add anything useful for that portion.

More and more, what I'm realizing is that the efforts that have made ARM what it is today have been good, but have not focused on what I believe most of us REALLY need... a direct WebUI for MakeMKV with a few extra tools added in.

**Reply by mihawk90** [reply]
> So yes, it isn't automatic, but that is at odds with the intended use case. :)

The intended use case is ripping tracks. ARM is not a media organiser and never claimed to be.
There are some things bolted on that try to make some educated guesses but that is not the primary use.

> And file names are numbered starting at zero which doesn't when you're trying to rip Season 06 Episodes 07 through 12...

The numbers aren't assigned by ARM. They are assigned by MakeMKV when it gets the track listing. And if you don't like the numbers starting from 0, then [simply change it](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/MakeMKV-Info).

---

Regardless, I think there are some wildly misplaced expectations of what ARM - or any actual media organiser for that matter - can do on an automated basis.

And this has also been [discussed at length before](https://github.com/automatic-ripping-machine/automatic-ripping-machine/discussions/724).

MakeMKV does not provide any episode info because they are not on the Disc. The only place on Disc where episodes are labeled as such is in the menu and MakeMKV cannot read those or rather cannot assign them directly to a title. *If* that were possible then that is a feature request for MakeMKV, not ARM. But since MakeMKV provides no such information, how is ARM supposed to know about it?

As discussed in the thread linked above the only way to reliably assign Episodes to Playlists on the Disc is by querying a database that links the two, and such a database simply does not exist. You're welcome to start a project for such.

**Reply by gibby82** [reply]
@ember1205 I've noticed that OMDB just doesn't pick things up nearly as well as TVDB. I have an API key setup for both. 

Agreed regarding Filebot - the workflow is typically rip or rip/compress, then rename to match basic season and episode templates, then run through Filebot against TVDB. 

I do agree - perhaps another approach is a good idea. 

@mihawk90 You are right, it is a 'ripping' machine. I guess I would say if it or MakeMKV can't get basic episode or track order right to enable simple organization efforts afterward then that's where my problem is. In instances where the order and info is correct it's easy to work around. If it's not, then it just isn't worth the effort IMO. If it's because the disc info sucks, then it is what it is I guess. 

I did ask in my first reply if there was an existing thread, so I'd ask for some leeway there.

**Reply by mihawk90** [reply]
> If it's because the disc info sucks, then it is what it is I guess.

It is, because DVD and BD were not designed with ripping in mind (which I don't think needs to be said), and from a technical standpoint players don't have a need for them.
From a specification perspective the formats were designed with a human-interactable menu in mind, which internally links the menu to a specific playlist. But what that playlist is - be it an episode, a trailer, an extra, a soundtrack, whatever - is not relevant to the player and therefore not included.

DVD menus are fairly simplistic so at some point MakeMKV added some smart(-ish) logic to "walk" through the menu and it can number the files accordingly. This actually tends to works somewhat well if the menu isn't very complex. If it can't walk through the menu for one reason or another it just falls back to playlist numbering.

But for BD that doesn't exist. On the one hand BD menus are more complex to begin with, on the other hand they are also everchanging by design. The BD menus are essentially Java applets, and in addition to that there's the whole BD Plus rabbithole (though I haven't seen that on TV Show BDs). But going back to the problem, BD menus are more freeform design whereas DVDs are a pretty rigid structure, so a "walking the menu" type deal isn't easily done. Even if it was, it's questionable how useful that would actually be with copy protections outside of the menu.

---

On another side note: For me sorting the filenames was not that much of a deal. I set MakeMKV to produce filenames like `DiscTitle_S99E<titleID>.mkv`. MakeMKV already produces a directory per Disc and unless the Disc titles are the same, that means each Disc will have its own directory. From that point on I can just add each directory individually to a mediamanager, change the Season (which will be parsed as 99), and then go through the files to assign the episode numbers. It's still manual obviously, but at least I don't need to do the naming manually.

However, the real "fun" started sorting through the audio and subtitle tracks. As with every other information, what each track contains is irrelevant to the player - beyond the language flag (though I've come across *one* BD that didn't have them set properly), so a track being commentary for example is something you'd just have to figure out on your own. For one series in particular I also ended up with a bunch of ripped but empty subtitle tracks, and forced-but-not-really-forced subtitle tracks, and sorting through *that* took me a good week or two for the series.
Ended up writing some scripts to automate some of this.

**Reply by ember1205** [reply]
| The numbers aren't assigned by ARM. They are assigned by MakeMKV when it gets the track listing

A) I know this isn't set by ARM
B) It doesn't really matter where the starting number is set or what it's set to - the point is that the numbering isn't going to align with ripping TV episides.

**Reply by mihawk90** [reply]
> B) It doesn't really matter where the starting number is set or what it's set to - the point is that the numbering isn't going to align with ripping TV episides.

Sure, but you're commenting on a discussion in the ARM repo about ARM having issues identifying/naming/whatever TV episodes. And my point simply was that ARM can't reasonably do anything about that because it - just like you - has no way to tell what number aligns with what episode.
Even a database mapping the numbers to shows and episodes would not work because the numbers can differ between runs and also depending on min/max length settings set by users. What *could potentially* work is linking the *playlist* numbers to episodes, but as I said there isn't a database for this (well, [sort of](https://github.com/automatic-ripping-machine/automatic-ripping-machine/issues/1344)). Plus, that would also require changing the filename template for MakeMKV.

**Reply by ember1205** [reply]
> Sure, but you're commenting on a discussion in the ARM repo about ARM having issues identifying/naming/whatever TV episodes. 

No. I'm commenting on a discussion in the ARM repo about why it's hard to identify TV Episodes. In general. Regardless of whether you're direct ripping via MakeMKV or using ARM in front of it. 

You need to stop presuming that every comment made here pertains ONLY to ARM and is some sort of attack on the product.

**Comment by HaroldRobi** [comment]
Yes unfortunately ARM does not have that capability.  I have 18 seasons of
"Murdoch Mysteries" I have to go through.  This series does not even
identify the episodes within the specific episode, like was done with
"Murder She Wrote".  Oh woes me  I have to watch at least 10 minutes of the
episode just to be able to identify it correctly.  That is really a pain
the the a** but I will get through it.  One episode at a time.

On Sat, Feb 7, 2026 at 7:22 AM ember1205 ***@***.***> wrote:

> Sure, but you're commenting on a discussion in the ARM repo about ARM
> having issues identifying/naming/whatever TV episodes.
>
> No. I'm commenting on a discussion in the ARM repo about why it's hard to
> identify TV Episodes. In general. Regardless of whether you're direct
> ripping via MakeMKV or using ARM in front of it.
>
> You need to stop presuming that every comment made here pertains ONLY to
> ARM and is some sort of attack on the product.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/automatic-ripping-machine/automatic-ripping-machine/discussions/594#discussioncomment-15727298>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUCFETJKDIJB4RFLVAQTL4L4KX7NBAVCNFSM6AAAAACQGM6YW6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTKNZSG4ZDSOA>
> .
> You are receiving this because you were mentioned.Message ID:
> <automatic-ripping-machine/automatic-ripping-machine/repo-discussions/594/comments/15727298
> @github.com>
>


-- 
-------------
"We've all seen that bumper sticker:    If God rode a motorcycle, it would
be a Harley.  I thought to myself...... now I know why God
hasn't made His second coming.....  He's still stuck in a cornfield out in
Nebraska or somewhere waiting for parts!"

**Reply by mihawk90** [reply]
> This series does not even identify the episodes within the specific episode

I've had that with House. If the show shows guest stars at the beginning or in the end credits you can probably skip a lot of the viewing if you just open up the episode listing on TMDB or wherever and check the guest stars.

**Comment by HaroldRobi** [comment]
You know that is a fantastic idea. I will give that a try.  Thank you

-------------
"We've all seen that bumper sticker:    If God rode a motorcycle, it would
be a Harley.  I thought to myself...... now I know why God
hasn't made His second coming.....  He's still stuck in a cornfield out in
Nebraska or somewhere waiting for parts!"
